CMakeLists.txt

```
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 3.22)
+
+project(ClpIrV1Decoder)
+
+# Download and extract Boost
+set(BOOST_VERSION "1.85.0")
+string(REPLACE "." "_" BOOST_ARCHIVE_VERSION_PART ${BOOST_VERSION})
+
+include(ExternalProject)
+ExternalProject_Add(
+        Boost
+        URL "https://archives.boost.io/release/${BOOST_VERSION}/source/boost_${BOOST_ARCHIVE_VERSION_PART}.tar.gz"
+        PREFIX "${CMAKE_BINARY_DIR}/boost"
+        CONFIGURE_COMMAND ""
+        BUILD_COMMAND ""
+        INSTALL_COMMAND ""
+        LOG_DOWNLOAD ON
+)
+ExternalProject_Get_Property(Boost SOURCE_DIR)
+set(BOOST_EXTRACT_DIR "${SOURCE_DIR}")
```

Looking online it seems a better alternative is to use `FetchContent_Declare` + `FetchContent_MakeAvailable`. 

Let's place `include()` at the top of the file just as how we include headers in source files in source codes. (See the top for the suggested change. )

```suggestion
set(BOOST_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/boost" CACHE STRING
    "Directory containing Boost sources, fetched from official release if non existent.")
if (NOT EXISTS ${BOOST_SRC_DIR})
    FetchContent_Declare(
        Boost
        URL "https://archives.boost.io/release/${BOOST_VERSION}/source/boost_${BOOST_ARCHIVE_VERSION_PART}.tar.gz"
        SOURCE_DIR ${BOOST_SRC_DIR}
    )
    message(STATUS "Fetching Boost from official release.")
    FetchContent_MakeAvailable(Boost)
endif()
```